### PR TITLE
[SPARK-33156][INFRA] Upgrade GithubAction image from 18.04 to 20.04

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to upgrade `Github Action` runner image from `Ubuntu 18.04 (LTS)` to `Ubuntu 20.04 (LTS)`.

### Why are the changes needed?

`ubuntu-latest` in `GitHub Action` is still `Ubuntu 18.04 (LTS)`.
- https://github.com/actions/virtual-environments#available-environments

This upgrade will help Apache Spark 3.1+ preparation for vote and release on the latest OS.

This is tested here.
- https://github.com/dongjoon-hyun/spark/pull/36

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the `Github Action` in this PR.

### CODE REVIEW

**Review by HyukjinKwon — COMMENTED** [review]
Seems like `ubuntu-20.04` is a preview one and that's why `ubuntu-latest` doesn't point this out (I read https://github.com/actions/virtual-environments/issues/1430 and https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources).

Should we just maybe leave it as is so GitHub Actions will updates it in their side? In this way, we could avoid updating this version later when another Ubuntu comes out.

**Comment by dongjoon-hyun** [comment]
The reason why they are still there officially is that they couldn't verify that how many plugins (or actions) in their market place fail to support it.

However, it's not true for us. I verified that we don't use none of the problematic actions in dongjoon-hyun#36 .

And, `ubuntu_20.04` is LTS. I don't think we need to upgrade this until next LTS (22.04). It's almost two years.

**Comment by HyukjinKwon** [comment]
Okay, I don't have a strong opinion here. cc @maropu or @srowen

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
   # Build: build Spark and run the tests for specified modules.
   build:
     name: "Build modules: ${{ matrix.modules }} ${{ matrix.comment }} (JDK ${{ matrix.java }}, ${{ matrix.hadoop }}, ${{ matrix.hive }})"
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (maropu) [inline .github/workflows/build_and_test.yml:20]: nit: how about leaving comments about the next LTS of ubuntu-20.04 here?

**Comment by maropu** [comment]
Seems fine to me

**Review by maropu — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @maropu . Sure!

**Comment by dongjoon-hyun** [comment]
Today, `Scala Repo` is extremely unstable. I saw the failure in all CIs.
```
Error:  https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.eed3si9n/sbt-assembly/scala_2.12/sbt_1.0/0.15.0/jars/sbt-assembly.jar: download error: Caught java.io.IOException: Server returned HTTP response code: 504 for URL: https://dl.bintray.com/sbt/sbt-plugin-releases/com.eed3si9n/sbt-assembly/scala_2.12/sbt_1.0/0.15.0/jars/sbt-assembly.jar (Server returned HTTP response code: 504 for URL: https://dl.bintray.com/sbt/sbt-plugin-releases/com.eed3si9n/sbt-assembly/scala_2.12/sbt_1.0/0.15.0/jars/sbt-assembly.jar) while downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.eed3si9n/sbt-assembly/scala_2.12/sbt_1.0/0.15.0/jars/sbt-assembly.jar
```

**Comment by dongjoon-hyun** [comment]
Github Action already passed in the previous commit and the last one is only adding comment. I'll merge this.
Thank you so much, @HyukjinKwon and @maropu !
Merged to master.

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (maropu) [reply .github/workflows/build_and_test.yml:21]: IMO `latest` was a bit ambiguous, so the new one (`ubuntu-20.04`)  can make it easier to understand in which environment we run tests. Thanks, @dongjoon-hyun !

**Pull request closed without merging**